### PR TITLE
Add highlights to the v2.10 changelog

### PR DESCRIPTION
**Pull request by bouweandela**
## Description

Add highlights to the v2.10.0 changelog.

Link to documentation: https://esmvaltool--2282.org.readthedocs.build/projects/ESMValCore/en/2282/changelog.html#v2-10-0

***

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: doc/changelog.rst [hunk 1]
```diff
@@ -9,7 +9,24 @@ v2.10.0
 -------
 Highlights
 
-TODO: add highlights
+-  All statistics preprocessors support the same operators and have a common
+   `documentation <stat_preprocs>`__. In addition, arbitrary keyword arguments
```
Comment (schlunma) [inline doc/changelog.rst:13]: This link does not work for me

File: doc/changelog.rst [hunk 2]
```diff
@@ -9,7 +9,24 @@ v2.10.0
...
+
+-  The output webpage generated by the tool now looks better and provides
+   methods to select and filter the output.
+
+-  Improved computational efficiency:
+
+   -  Automatic rechunking between preprocessor steps to keep the
+      `graph size smaller <https://docs.dask.org/en/latest/best-practices.html#avoid-very-large-graphs>`_
+      and the `chunk size optimal <https://blog.dask.org/2021/11/02/choosing-dask-chunk-sizes>`__.
+   -  Reduce the size of the dask graph created by :func:`esmvalcore.preprocessor.anomalies`
```
Comment (schlunma) [inline doc/changelog.rst:24]:
```suggestion
   -  Reduce the size of the dask graph created by :func:`esmvalcore.preprocessor.anomalies`.
```

File: doc/changelog.rst [hunk 3]
```diff
@@ -9,7 +9,24 @@ v2.10.0
...
+   -  Preprocessors :func:`esmvalcore.preprocessor.mask_above_threshold`,
+      :func:`esmvalcore.preprocessor.mask_below_threshold`,
+      :func:`esmvalcore.preprocessor.mask_inside_range`,
+      :func:`esmvalcore.preprocessor.mask_outside_range` are now lazy
```
Comment (schlunma) [inline doc/changelog.rst:28]:
```suggestion
      :func:`esmvalcore.preprocessor.mask_outside_range` are now lazy.
```

File: doc/changelog.rst [hunk 4]
```diff
@@ -9,7 +9,24 @@ v2.10.0
...
+   -  Lazy coordinates bounds are no longer loaded into memory by the CMOR checks and fixes
```
Comment (schlunma) [inline doc/changelog.rst:29]:
```suggestion
   -  Lazy coordinates bounds are no longer loaded into memory by the CMOR checks and fixes.
```

**Review by schlunma — COMMENTED** [review]
Thanks Bouwe, looks good to me. I have some minor comments.

One comment not directly related to this PR: to me it's not 100% clear how we distinguish between "Backwards incompatible changes" and "Deprecations".  I think some entries in "Backwards incompatible changes" would rather belong to "Deprecations".

**Comment by bouweandela** [comment]
> t's not 100% clear how we distinguish between "Backwards incompatible changes" and "Deprecations"

"Deprecations" are "Backwards incompatible changes" that are planned for a future release. Does that clarify it?

**Comment by schlunma** [comment]
> "Deprecations" are "Backwards incompatible changes" that are planned for a future release. Does that clarify it?

Agreed. But then 

> Added new operators for statistics preprocessor (e.g., 'percentile') and allowed passing additional arguments ([#2191](https://github.com/ESMValGroup/ESMValCore/pull/2191)) [Manuel Schlund](https://github.com/schlunma)
> - This harmonizes the operators for all statistics preprocessors. From this version, the new names can be used; the old arguments will stop working from version 2.12.0. Please refer to [Statistical preprocessors](https://esmvaltool--2282.org.readthedocs.build/projects/ESMValCore/en/2282/recipe/preprocessor.html#stat-preprocs) for a detailed description.

needs to got to "Deprecations"

File: doc/changelog.rst (hunk 1 above)
Comment (bouweandela) [reply doc/changelog.rst]: Hopefully fixed in [f79648b](https://github.com/ESMValGroup/ESMValCore/pull/2282/commits/f79648b2e609e4c3247ce88a014922dfe335ceb6)

**Review by schlunma — APPROVED** [review]
Thanks 🚀

**Pull request merged**